pkg/tests/apis/provisioning/provisioning_test.go

```diff
@@ -608,7 +612,29 @@ func TestProvisioning_ExportUnifiedToRepository(t *testing.T) {
 	// And time to assert.
 	helper.AwaitJobs(t, repo)
 
-	fpath := filepath.Join(helper.ProvisioningPath, slugify.Slugify(mustNestedString(dashboard.Object, "spec", "title"))+".json")
-	_, err = os.Stat(fpath)
-	require.NoError(t, err, "exported file was not created at path %s", fpath)
+	// Check that each file was exported with its stored version
+	for _, title := range []string{
+		"Test dashboard. Created at v1",
+		"Test dashboard. Created at v2",
+	} {
+		fpath := filepath.Join(helper.ProvisioningPath, slugify.Slugify(title)+".json")
+		//nolint:gosec // we are ok with reading files in testdata
+		body, err := os.ReadFile(fpath)
+		require.NoError(t, err, "exported file was not created at path %s", fpath)
+		obj := map[string]any{}
+		err = json.Unmarshal(body, &obj)
+		require.NoError(t, err, "exported file not json %s", fpath)
+		apiVerison, _, err := unstructured.NestedString(obj, "apiVersion")
+		require.NoError(t, err)
+		name, _, err := unstructured.NestedString(obj, "metadata", "name")
+		require.NoError(t, err)
+		switch name {
```

Comment:
let's make part of the test spec and have a different test per version / case instead of having a single test with a switch.